src/app/components/ManualEntry.svelte

```diff
@@ -84,50 +132,87 @@
     font: 1em var(--main-font);
   }
 
-  input:focus {
+  table input:focus {
     background-color: var(--lite-white);
   }
+
+  table p {
+    display: inline;
+    color: var(--red);
+  }
+
+  .save-zone {
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    justify-content: center;
+    padding: 1%;
+  }
 </style>
 
 <div class="language__sources-manual-entry">
   <div class="language__sources-manual-entry-tablename">
     <h4>Table Name</h4>
-    <input data-cy="manual-entry-input-tablename" />
+    <input
+      type="text"
+      bind:value={tableData.name}
+      data-cy="manual-entry-input-tablename" />
   </div>
 
-  <table>
+  <table id="manual-entry-table">
     <thead>
-      <th>Word</th>
-      <th>Count</th>
-      <th>Action</th>
+      {#each columns as column (column.name)}
+        <th>{column.label}</th>
+      {/each}
+      <th />
     </thead>
 
+    {#each rowDataFromManualEntry as row, i (i)}
+      <tr>
+        <td>
+          <input
+            type="text"
+            bind:value={row.word}
+            data-cy="manual-entry-input-word" />
+          {#if !validateRowData(row)}
+            <p>*</p>
+          {/if}
+        </td>
+        <td>
+          <input
+            type="number"
+            min="0"
+            placeholder="Optional"
+            bind:value={row.count}
+            data-cy="manual-entry-input-count" />
+        </td>
+        <td>
+          <button
+            class="btn--inline"
+            on:click={() => deleteRow(i)}
+            data-cy="manual-entry-delete">Delete</button>
+        </td>
+      </tr>
+    {/each}
     <tr>
-      <td><input type="text" data-cy="manual-entry-input-word" /></td>
-      <td>
-        <input
-          type="number"
-          placeholder="Optional"
-          data-cy="manual-entry-input-count" />
-      </td>
-      <td>
+      <td colspan="3">
         <button
           class="btn--inline"
-          on:click={deleteRow}
-          data-cy="manual-entry-delete">Delete</button>
+          on:click={addNewRow}
+          data-cy="munual-entry-add-row">Add Row</button>
       </td>
     </tr>
-    <!-- TODO: hard code new row for now, but we can auto generate new row later -->
-    <tr>
-            <td>
-                <input type="text" />
-            </td>
-            <td>
-                <input type="number" placeholder="Optional" />
-            </td>
-            <td>
-                <button class="btn--inline" on:click={deleteRow}>Delete</button>
-            </td>
-        </tr>
   </table>
+
+  <div class="save-zone">
+    <Button
+      color={btnColor}
+      isOutlined
+      disabled={!validInput}
```

Comment:
I do not like this solution (having a manual save button), as I think an auto-save option is better. However, I think autosave is enough work that it's a new PR 😅 

The problem with a button that gets disabled on invalid input is that there's no written feedback as to what exactly is wrong with the form. "Why can't I save"? The `required` attribute of `<input>` would help with this. Note: you may want to wrap the entire table in `<form>` element, and then make the underlying button here be a `<button type="submit">`. This will enable the browser to automatically validate the `required` attribute AND tell you which words are missing!